lib/rubocop/ast/processed_source.rb

```diff
@@ -397,6 +397,31 @@ def create_parser(ruby_version, parser_engine, prism_result)
       end
       # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
 
+      def normalize_parser_engine(parser_engine, ruby_version)
+        parser_engine = parser_engine.to_sym
+        unless PARSER_ENGINES.include?(parser_engine)
+          raise ArgumentError, 'The keyword argument `parser_engine` accepts `default`, ' \
+                               "`parser_whitequark`, or `parser_prism`, but `#{parser_engine}` " \
+                               'was passed.'
+        end
+        if parser_engine == :default
+          best_parser_engine(ruby_version)
+        else
+          parser_engine
+        end
+      end
+
+      # The Parser gem does not support Ruby 3.5 or later.
+      # It is also not fully compatible with Ruby 3.4 but for
+      # now respects using parser for backwards compatibility.
+      def best_parser_engine(ruby_version)
```

Comment:
Can you rename it to the following?

```suggestion
      def default_parser_engine(ruby_version)
```